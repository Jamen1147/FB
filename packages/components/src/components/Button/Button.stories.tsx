import Button from './Button';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'Button',
  component: Button,
} as Meta;

const Template: Story<React.ComponentProps<typeof Button>> = (args) => (
  <Button {...args}>Button</Button>
);

export const Default = Template.bind({});
