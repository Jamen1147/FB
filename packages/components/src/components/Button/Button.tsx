import { ComponentProps, PropsWithChildren } from 'react';
import { Button as MUIButton } from '@mui/material';

const Button = ({
  children,
  ...props
}: PropsWithChildren<ComponentProps<typeof MUIButton>>) => (
  <MUIButton {...props}>{children}</MUIButton>
);

export default Button;
