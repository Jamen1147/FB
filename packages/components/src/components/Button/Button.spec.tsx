import Button from './Button';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('<Button />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Button>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('invokes function on click', () => {
    const btnText = 'Button';
    const onClick = jest.fn();

    render(<Button onClick={onClick}>{btnText}</Button>);

    userEvent.click(screen.getByText(btnText));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
