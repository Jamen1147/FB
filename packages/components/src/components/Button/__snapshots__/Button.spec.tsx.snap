// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Button
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
